azure-functions-powershell-library/WorkerIndexingHelper.cs

```diff
@@ -0,0 +1,210 @@
+
+            thisFunction.FunctionId = Guid.NewGuid().ToString();
+
+            ParamBlockAst paramBlock = (ParamBlockAst)potentialFunction.Find(x => x is ParamBlockAst, true);
+            ExtractBindings(thisFunction, paramBlock);
+
+            return thisFunction;
+        }
+
+        private static void ExtractBindings(FunctionInformation thisFunction, ParamBlockAst paramBlock)
+        {
+            if (paramBlock == null)
+            {
+                return;
+            }
+
+            IEnumerable<AttributeAst>? functionAttribute = paramBlock.Attributes.Where(x => x.TypeName.Name == "Function" && x.PositionalArguments.Count > 0);
+            if (functionAttribute.Any() && functionAttribute.First().PositionalArguments[0].GetType() == typeof(StringConstantExpressionAst))
+            {
+                thisFunction.Name = ((StringConstantExpressionAst)functionAttribute.First().PositionalArguments[0]).Value;
+            }
+
+            List<BindingInformation> inputBindings = GetInputBindingInfo(paramBlock);
+            List<BindingInformation> outputBindings = GetOutputBindingInfo(paramBlock.Attributes);
+            List<BindingInformation> missingBindings = BindingExtractor.GetDefaultBindings(inputBindings, outputBindings);
+
+            thisFunction.Bindings.AddRange(inputBindings);
+            thisFunction.Bindings.AddRange(outputBindings);
+            thisFunction.Bindings.AddRange(missingBindings);
+        }
+        private static List<BindingInformation> GetInputBindingInfo(ParamBlockAst paramBlock)
+        {
+            List<BindingInformation> outputBindingInfo = new List<BindingInformation>();
+            foreach (ParameterAst parameter in paramBlock.Parameters)
+            {
+                foreach (AttributeAst attribute in parameter.Attributes)
+                {
+                    BindingInformation? bindingInfo = BindingExtractor.ExtractInputBinding(attribute, parameter);
+                    if (bindingInfo is not null)
+                    {
+                        outputBindingInfo.Add(bindingInfo);
+                    }
+                }
+            }
+            return outputBindingInfo;
+        }
+
+        private static List<BindingInformation> GetOutputBindingInfo(ReadOnlyCollection<AttributeAst> attributes)
+        {
+            List<BindingInformation> outputBindingInfo = new List<BindingInformation>();
+            foreach (AttributeAst attribute in attributes)
+            {
+                BindingInformation? bindingInformation = BindingExtractor.ExtractOutputBinding(attribute);
+                if (bindingInformation is not null)
+                {
+                    outputBindingInfo.Add(bindingInformation);
+                }
+            }
+            return outputBindingInfo;
+        }
+
+        // Everything below this point are static methods that may be used by classes implementing IBinding to extract information from 
```

Comment:
No strong preference on my end. I think that, since this class is also a "helper"/utility class (in the name), it's not wrong to include some static utility methods that other files may reference. In short - I'm actually okay with keeping these methods here. How do others feel?

Reply:
Leaving this for now, we can revisit later if it becomes an issue